makefiles/instance/rules.mk

```diff
@@ -292,22 +292,20 @@ $(THEOS_OBJ_DIR)/%.xm.$(_THEOS_OBJ_FILE_TAG).o: %.xm $(THEOS_OBJ_DIR)/%.xm.mm
 $(THEOS_OBJ_DIR)/%.mi: %.xi
 	$(ECHO_NOTHING)mkdir -p $(dir $@)$(ECHO_END)
 	$(ECHO_PREPROCESSING)$(ECHO_UNBUFFERED)$(TARGET_CXX) -x objective-c -E -I"$(call __clean_pwd,$(dir $<))" $(_THEOS_INTERNAL_IFLAGS_C) $(ALL_DEPFLAGS) $(ALL_PFLAGS) $(PREPROCESS_ARCH_FLAGS) $(ALL_OBJCFLAGS) $< > $(THEOS_OBJ_DIR)/$<.pre && $(THEOS_BIN_PATH)/logos.pl $(ALL_LOGOSFLAGS) $(THEOS_OBJ_DIR)/$<.pre > $(THEOS_OBJ_DIR)/$<.mi $(ECHO_END)
-	$(ECHO_NOTHING)$(THEOS_BIN_PATH)/logos.pl $(ALL_LOGOSFLAGS) $< > $@$(ECHO_END)
```

Comment:
Would we want to clean up the temporary `$(THEOS_OBJ_DIR)/$<.pre`?

I was going to suggest piping to logos.pl but it doesn't take stdin as input (and adding support for it might be complicated as it performs seeks on the input file)

Reply:
It doesn't look like the .pre's are used anywhere other than the logos calls, so removing them should be alright. Just say the word and I can either add them to the logos intermediate removal command or add a new removal under the rule where they're used.

Reply:
I'd say to add them where they're used (replacing the lines you removed in this PR). I'll try to add support for logos to support stdin as input and then we can pipe through without the intermediate files

Reply:
Done. Also simplified the other removals with `$<` (thanks for that!)